internal/services/bot/bot_healthbot_resource.go

```diff
@@ -161,8 +162,11 @@ func resourceHealthbotServiceUpdate(d *pluginsdk.ResourceData, meta interface{})
 		payload.Tags = tags.Expand(d.Get("tags").(map[string]interface{}))
 	}
 
-	if _, err := client.BotsUpdate(ctx, *id, payload); err != nil {
-		return fmt.Errorf("updating %s: %+v", id, err)
+	if resp, err := client.BotsUpdate(ctx, *id, payload); err != nil {
+		// update check logic once the issue https://github.com/Azure/azure-rest-api-specs/issues/22791 is fixed
+		if !response.WasStatusCode(resp.HttpResponse, http.StatusCreated) {
+			return fmt.Errorf("updating %s: %+v", id, err)
+		}
```

Comment:
If this is a 201 that would indicate that this is an LRO, so there should be something [we can poll on using the pollers within `go-azure-sdk` here](https://github.com/hashicorp/go-azure-sdk/blob/6cdea3ca229b352fa45cf3efcc2ece9b0eaca92a/sdk/client/resourcemanager/poller.go#L12)?

Reply:
As service team has submitted this [PR](https://github.com/Azure/azure-rest-api-specs/pull/22908) to fix this issue but it hasn't been merged, so I intend to update this PR accordingly once this PR is merged.

Reply:
@neil-yechenwei - the pr linked here has been merged

Reply:
Thanks @katbyte for reminder. I updated PR. 

![image](https://user-images.githubusercontent.com/19754191/229400744-de2a9eaa-6587-4a62-9411-ca2107f2b3b5.png)

Reply:
@katbyte , kindly ping. Is it fine to merge this PR?